Source/ASNodeController+Beta.mm

```diff
@@ -12,33 +12,33 @@
 #import <AsyncDisplayKit/ASNodeController+Beta.h>
 #import <AsyncDisplayKit/ASThread.h>
 
+#import <mutex>
+
 #define _node (_shouldInvertStrongReference ? _weakNode : _strongNode)
 
 @implementation ASNodeController
 {
   ASDisplayNode *_strongNode;
   __weak ASDisplayNode *_weakNode;
-  ASDN::RecursiveMutex __instanceLock__;
+  std::once_flag _nodeOnce;
 }
 
-- (void)loadNode
+- (ASDisplayNode *)createNode
 {
-  ASLockScopeSelf();
-  self.node = [[ASDisplayNode alloc] init];
+  return [[ASDisplayNode alloc] init];
 }
 
 - (ASDisplayNode *)node
 {
-  ASLockScopeSelf();
-  if (_node == nil) {
-    [self loadNode];
-  }
+  std::call_once(_nodeOnce, [self](){
+    [self setupReferencesWithNode:[self createNode]];
```

Comment:
Since we can't enforce that createNode is not called externally (or at least don't now), is it possible to only call it here if it hasn't been called yet / _node is nil?